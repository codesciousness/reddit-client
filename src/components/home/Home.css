.Home__trending, .Home__latest {
    width: 95%;
    max-width: 1800px;
    margin: 3rem auto;
}

.Home__divider__container {
    width: 90%;
    margin: 2rem auto;
    position: relative;
}

.Home__divider__container hr {
    border-bottom: 1px solid black;
}

.Home__divider {
    display: inline-block;
    position: absolute;
    top: -50%;
    left: 50%;
    transform: translate(-50%, -50%);
    font-weight: bold;
    text-align: center;
    text-transform: uppercase;
    color: white;
    background: linear-gradient(to right, rgb(255, 2, 108), purple);
    padding: 0.25rem;
}