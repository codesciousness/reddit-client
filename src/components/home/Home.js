import React from 'react';
import SearchTerm from '../../features/searchTerm/SearchTerm';
import Posts from '../../features/posts/Posts';

const Home = () => {
    return (
        <main>
          <SearchTerm />
          <section className="App__trending">
            <div className="App__divider__container">
              <hr />
              <div className="App__divider">Trending</div>
            </div>
            <Posts isTrending={true} />
          </section>
          <section className="App__latest">
            <div className="App__divider__container">
              <hr />
              <div className="App__divider">Latest</div>
            </div>
            <Posts isTrending={false} />
          </section>
        </main>
    );
}

export default Home;