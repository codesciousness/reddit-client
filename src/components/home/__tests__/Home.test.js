import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import Home from '../Home';

jest.mock('../../../features/searchTerm/SearchTerm', () => () => (<form data-testid="SearchTerm"><input placeholder="Test Search..."/></form>));
jest.mock('../../../features/posts/Posts', () => ({ isTrending }) => {
    if(isTrending){
        return <section data-testid="Posts__trending">Test Trending Posts</section>
    }
    else {
        return <section data-testid="Posts__latest">Test Latest Posts</section>
    }
});

describe('<Home />', () => {
    test('Home component is defined', () => {
        expect(<Home />).toBeDefined();
    });
    test('matches the snapshot', () => {
        const component = create(<Home />);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly', () => {
        render(<Home />);
        expect(screen.getByText('Trending')).toBeInTheDocument();
        expect(screen.getByText('Latest')).toBeInTheDocument();
    });
    test('renders a SearchTerm component', () => {
        render(<Home />);
        expect(screen.getByTestId('SearchTerm')).toBeInTheDocument();
        expect(screen.getByPlaceholderText(/Test Search/)).toBeInTheDocument();
    });
    test('renders 2 Posts components', () => {
        render(<Home />);
        expect(screen.getAllByText(/Posts/)).toHaveLength(2);
    });
    test('passes correct props to subcomponents', () => {
        render(<Home />);
        expect(screen.getByTestId('Posts__trending')).toBeInTheDocument();
        expect(screen.getByTestId('Posts__latest')).toBeInTheDocument();
    });
});