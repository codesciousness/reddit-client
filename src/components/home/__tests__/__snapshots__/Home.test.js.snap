// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches the snapshot 1`] = `
<main>
  <form
    data-testid="SearchTerm"
  >
    <input
      placeholder="Test Search..."
    />
  </form>
  <section
    className="Home__trending"
  >
    <div
      className="Home__divider__container"
    >
      <hr />
      <div
        className="Home__divider"
      >
        Trending
      </div>
    </div>
    <section
      data-testid="Posts__trending"
    >
      Test Trending Posts
    </section>
  </section>
  <section
    className="Home__latest"
  >
    <div
      className="Home__divider__container"
    >
      <hr />
      <div
        className="Home__divider"
      >
        Latest
      </div>
    </div>
    <section
      data-testid="Posts__latest"
    >
      Test Latest Posts
    </section>
  </section>
</main>
`;
